test/generate-trace.sh

```diff
@@ -1,9 +1,17 @@
 #!/bin/bash
 
-tracegen -otlp-endpoint localhost:55680 -otlp-insecure -service e2e-test
+# we send first the no auth, so that we get higher assurance that the non-existence of the
+# span is because of the auth failure instead of network latency
+tracegen -otlp-endpoint localhost:55685 -otlp-insecure -service e2e-test-agent-without-auth &>> ./test/tracegen.log
 if [ $? != 0 ]; then
-    echo "Failed to generate a trace."
+    echo "Failed to generate a trace to the agent that does not authenticate with the collector."
     exit 1
 fi
 
-echo "✅ Trace generated."
+tracegen -otlp-endpoint localhost:55680 -otlp-insecure -service e2e-test-agent-with-auth &>> ./test/tracegen.log
```

Comment:
One more nit.  Can you use a different name for logs?  Otherwise we clobber the output of the first run.

Reply:
How does it look now?

Reply:
I'll merge it for now, but let me know if it can be improved further.